lib/protobuf/field/base_field.rb

```diff
@@ -93,6 +103,68 @@ def default_value
                            end
       end
 
+      def define_encode_to_stream!
+        if repeated? && packed?
+          ::Protobuf::Field::BaseFieldMethodDefinitions.define_repeated_packed_encode_to_stream_method!(self)
+        elsif repeated?
+          ::Protobuf::Field::BaseFieldMethodDefinitions.define_repeated_not_packed_encode_to_stream_method!(self)
+        elsif message? || type_class == ::Protobuf::Field::BytesField
+          ::Protobuf::Field::BaseFieldMethodDefinitions.define_bytes_encode_to_stream_method!(self)
+        elsif type_class == ::Protobuf::Field::StringField
+          ::Protobuf::Field::BaseFieldMethodDefinitions.define_string_encode_to_stream_method!(self)
+        else
+          ::Protobuf::Field::BaseFieldMethodDefinitions.define_base_encode_to_stream_method!(self)
+        end
+      end
+
+      def define_field_p!
```

Comment:
I think these `_p` methods would read a lot better with `_predicate` spelled out. I'll leave it up to you.